### ISSUE TITLE
Proposing new features: Developing Echo State Network layers

### ISSUE THREAD

**Issue by brian101bedrockAi** [pytorch/pytorch#51003]
Hello,

I would like to contribute to PyTorch with a new feature. I am currently working on research using ESNs and have written some ESN models in PyTorch for the GPU and CPU. I think these could be useful to the community.

I would initially propose a leaky-Integrator based ESN layer. From this I believe it would be generalizable to other node types. This layer's functionality would include:
- initialize the input matrix using user chosen parameters
- initalize the reservoir matrix with user chosen parameters and a user chosen spectral radius.
- provide a novel ability to choose the number of 2-attractor resevoir nodes in the ESN when the spectral radius is greater than 1.
- a mechanism for passing gradients past this layer during backpropagation since the initialized reservoir and input matrices are constant after initialization.

ESNs can act as a dynamical system embedding in a network and I think this presents some interesting possibilites for the research community.  

Thank you for your time.

**Comment by brian101bedrockAi** [comment]
Hi,
I wasn't sure what the protocol is so any guidance you can give as to what I info I need to provide you would be great. I also know there is a lot of work for you all to cover before getting to this,  judging by my notifications. 

Thank you for your time,
Brian Whiteaker
